crates/core/tedge_api/src/workflow/script.rs

```diff
@@ -347,6 +348,45 @@ impl AwaitHandlers {
     }
 }
 
+/// Define state transition on each iteration outcome
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct IterateHandlers {
+    pub on_next: GenericStateUpdate,
+    pub on_success: GenericStateUpdate,
+    pub on_error: Option<GenericStateUpdate>,
+}
+
+impl IterateHandlers {
+    pub fn new(
+        on_next: GenericStateUpdate,
+        on_success: GenericStateUpdate,
+        on_error: Option<GenericStateUpdate>,
+    ) -> Self {
+        Self {
+            on_next,
+            on_success,
+            on_error,
+        }
+    }
+
+    pub fn with_default(
+        mut self,
+        default: &DefaultHandlers,
+    ) -> Result<Self, WorkflowDefinitionError> {
+        if self.on_error.is_none() {
+            self.on_error.clone_from(&default.on_error)
+        }
+
+        if self.on_error.is_none() {
+            return Err(WorkflowDefinitionError::MissingState {
+                state: "on_success".to_string(),
+            });
+        }
```

Comment:
There is a typo:

```suggestion
        if self.on_success.is_none() {
            return Err(WorkflowDefinitionError::MissingState {
                state: "on_success".to_string(),
            });
        }
```

Better is to provide a sensible default, instead of raising an error:

```suggestion
        if self.on_success.is_none() {
           self.on_success = Some(GenericStateUpdate::successful())
        }
```

Reply:
Resolved by 1012f41c0069b302fb54aa249149fdee9ab20071